runtime/compiler/env/VMJ9.cpp

```diff
@@ -2309,18 +2309,39 @@ TR_MarkHotField::markHotField(J9Class * clazz, bool rootClass)
       J9ROMClass* romClass = TR::Compiler->cls.romClassOf((TR_OpaqueClassBlock*)clazz);
       J9UTF8* name = J9ROMCLASS_CLASSNAME(romClass);
       printf("%*smarked field as hot in class %.*s\n", depth(), " ", J9UTF8_LENGTH(name), J9UTF8_DATA(name));
-      }
-
-   *(UDATA *)((char *)clazz + offsetOfHotFields()) = noncoldWord | _bitValue;
-
-   return true;
+      }  
+
+      if (jitConfig->javaVM->memoryManagerFunctions->j9gc_hot_reference_field_required(jitConfig->javaVM)) {
+         if(clazz->totalInstanceSize <=512) {
+            if ((_slotIndex !=  clazz->hotField2.hotFieldOffset) &&
+               ((clazz->hotField1.hotFieldOffset == UDATA_MAX && _comp->getMethodHotness() >=warm) ||
+               (_comp->getMethodHotness() > clazz->hotField1.hotFieldMethodHotness && frequency > ((clazz->hotField1.hotFieldThreshold/10)*9)) ||
```

Comment:
what are the 9 and 10 in terms of magic numbers here and why the 'fudge' factor? This deserves a comment and possibly naming these constants in some nice way.

Reply:
The 9 and 10 factors are numbers that I have been playing with to see if they have any effects on how hot fields are set and are a mechanism to allow a frequency range for setting a hot field in an attempt to better ensure the right hot field is being set. I am testing with and without these factors now to see their effects.

Reply:
My understanding is to achieve sort of hysteresis effect so that hot fields are not changed too frequently in case we have more than 2 of them that are approximately same hotness.

Now if that is important or not it's subject to exploration, although I don't see an obvious problem with changing them too frequently.

Reply:
well if the factor is going to remain the numbers needs to pulled out into some meaningful variable(s) and probably add an feGetEnv to adjust them along with a comment with some of the justification above.